vector/src/main/java/im/vector/app/features/settings/devices/v2/VectorSettingsDevicesFragment.kt

```diff
@@ -151,6 +157,41 @@ class VectorSettingsDevicesFragment :
         }
     }
 
+    private fun initQrLoginView() {
+        if (!vectorFeatures.isReciprocateQrCodeLogin()) {
+            views.deviceListHeaderSignInWithQrCode.isVisible = false
+            views.deviceListHeaderScanQrCodeButton.isVisible = false
+            views.deviceListHeaderShowQrCodeButton.isVisible = false
+            return
+        }
+
+        views.deviceListHeaderSignInWithQrCode.isVisible = true
+        views.deviceListHeaderScanQrCodeButton.isVisible = true
+        views.deviceListHeaderShowQrCodeButton.isVisible = true
+
+        views.deviceListHeaderScanQrCodeButton.debouncedClicks {
+            navigator
+                    .openLoginWithQrCode(
+                            requireActivity(),
+                            QrCodeLoginArgs(
+                                    loginType = QrCodeLoginType.LINK_A_DEVICE,
+                                    showQrCodeByDefault = false,
+                            )
+                    )
+        }
+
+        views.deviceListHeaderShowQrCodeButton.debouncedClicks {
+            navigator
+                    .openLoginWithQrCode(
+                            requireActivity(),
+                            QrCodeLoginArgs(
+                                    loginType = QrCodeLoginType.LINK_A_DEVICE,
+                                    showQrCodeByDefault = true,
+                            )
+                    )
+        }
```

Comment:
Maybe extract to a sub method to avoid code duplication:
```kotlin
private fun navigateToQrCodeScreen(showQrCodeImmediately: Boolean) {
            navigator
                    .openLoginWithQrCode(
                            requireActivity(),
                            QrCodeLoginArgs(
                                    loginType = QrCodeLoginType.LINK_A_DEVICE,
                                    showQrCodeByDefault = showQrCodeImmediately,
                            )
                    )
}
```

Reply:
Or better you can expose it inside the `VectorSettingsDevicesViewNavigator` so that we can test it.

Reply:
Done (didn't have time to create a standalone navigator, will do when we start writing tests for this feature).

Reply:
The navigator was already in place for info, the idea was just to move the method inside it.